sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxGatewayStoreModel.java

```diff
@@ -507,7 +506,8 @@ public Mono<RxDocumentServiceResponse> processMessage(RxDocumentServiceRequest r
                     }
 
                     if (Exceptions.isThroughputControlRequestRateTooLargeException(dce)) {
-                        BridgeInternal.recordGatewayResponse(request.requestContext.cosmosDiagnostics, request, null, dce, globalEndpointManager);
+                        StoreResponseDiagnostics storeResponseDiagnostics = StoreResponseDiagnostics.createStoreResponseDiagnostics(dce);
+                        BridgeInternal.recordGatewayResponse(request.requestContext.cosmosDiagnostics, request, storeResponseDiagnostics, globalEndpointManager);
```

Comment:
do we need to check whether request.requestContext.cosmosDiagnostics != null? (similar to line 410 - 416)

Reply:
Good catch, worth adding a null check here to avoid any NPEs.